src/components/messaging-app.tsx

```diff
@@ -133,7 +141,8 @@ export const MessagingApp: FC = () => {
 
       if (
         !lockRefreshRef.current &&
-        conversations[selectedConversationPublicKey]
+        conversations[selectedConversationPublicKey] &&
+        initConversationKey === selectedConversationPublicKeyRef.current
```

Comment:
the main issue is that just reading `selectedConversationPublicKey` here will result in the value that this variable had when the function was called, so before we did all the async calls.

The solution here will be to either use `useRef` on top of the initial `selectedConversationPublicKey` variable that will guarantee that you will get the latest state value, or introduce some global variable somewhere (e.g. on `window`) that will be updated synchronously

Reply:
thank you for the great explanation!